### ISSUE TITLE
Uncaught TypeError: Cannot read property 'hasOwnProperty' of undefined

### ISSUE THREAD

**Issue by Domy** [atom/atom#9747]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.2.3
**System**: Mac OS X 10.11.1
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'hasOwnProperty' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88

TypeError: Cannot read property 'hasOwnProperty' of undefined
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88:43)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:78:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:67:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:30:12)
    at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/gutter-container-component.js:79:25)
    at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:224:39)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:302:28
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:177:9)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:3:61
```

### Commands

```
     -1:27.4.0 core:paste (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
  3x -1:26.5.0 core:save (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -1:19.8.0 core:paste (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -1:19.2.0 core:save (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -1:13.5.0 core:paste (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -1:13 core:save (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -1:06.7.0 core:paste (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -1:05.9.0 core:save (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -1:00.8.0 core:paste (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -1:00.0 core:save (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -0:53 core:paste (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -0:52.4.0 core:save (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
  2x -0:46.6.0 core:undo (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -0:44.2.0 core:paste (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -0:22.8.0 core:save (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
     -0:07.7.0 core:paste (atom-text-editor.editor.vim-mode.insert-mode.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "linter",
      "file-icons"
    ]
  },
  "editor": {
    "invisibles": {}
  }
}
```

### Installed Packages

```coffee
# User
.bin, vundefined
atom-beautify, v0.28.19
atom-jade, v0.3.0
atom-typescript, v7.11.1
autoclose-html, v0.19.0
color-picker, v2.0.13
emmet, v2.3.14
language-babel, v2.5.0
language-jade, v0.6.2
language-nginx, v0.6.1
linter-csslint, v1.1.0
linter-htmlhint, v0.2.1
linter-jscs, v3.2.0
linter-jshint, v1.2.1
minimap, v4.16.1
pigments, v0.19.0
project-manager, v2.6.5
react, v0.12.10
vim-mode, v0.63.0

# Dev
No dev packages
```

**Comment by 50Wliu** [comment]
Hi @Domy, a few questions for you:
1. Can you reproduce this in safe mode (`atom --safe`)?
2. What were you doing when this occurred?
3. Does this error occur every time?

**Comment by as-cii** [comment]
This seems similar to #9299. We've received a bunch of these lately but I am having a hard time reproducing it: it seems to be related to some state in `LineNumbersTileComponent` but I am not clear on what exactly is causing this. Any help on making this reproducible would be super appreciated! :sparkles: 

Also, I am thinking it could be worth to add an `assert` statement, so that we can collect more information on the surrounding context when this error gets triggered.

@50Wliu: in the meantime, what do you think about closing this and reopening the aforementioned issue?

**Comment by 50Wliu** [comment]
@as-cii Sounds good!

@Domy: Please add the answers to my questions on #9299 instead.

**Comment by BenHand** [comment]
note: came across this error when pairing with another developer and pulling down fresh changes to the current file i was working in (no merge conflict). Error appeared immediately after the content of the file updated from a merge. 

**Atom Version**: 1.3.3
**System**: Mac OS X 10.11.2
**Thrown From**: Atom Core


### Stack Trace

Uncaught TypeError: Cannot read property 'hasOwnProperty' of undefined

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88

TypeError: Cannot read property 'hasOwnProperty' of undefined
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateLineNumbers (/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:88:43)
    at LineNumbersTileComponent.module.exports.LineNumbersTileComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/line-numbers-tile-component.js:78:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateTileNodes (/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:67:19)
    at LineNumberGutterComponent.module.exports.TiledComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/tiled-component.js:30:12)
    at GutterContainerComponent.module.exports.GutterContainerComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/gutter-container-component.js:79:25)
    at TextEditorComponent.module.exports.TextEditorComponent.updateSync (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:220:39)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor-component.js:298:28
    at ViewRegistry.module.exports.ViewRegistry.performDocumentUpdate (/Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:190:9)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/view-registry.js:3:61
```

### Commands

```
     -7:09.1.0 core:move-left (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
     -6:59.1.0 core:backspace (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
     -6:58.1.0 core:move-right (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
  2x -6:57.7.0 core:backspace (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
     -6:53.8.0 core:move-right (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
  2x -6:53.5.0 core:backspace (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
     -6:49.1.0 core:move-right (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
  2x -6:48.7.0 core:backspace (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
     -6:44.2.0 core:move-right (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
  5x -6:43.9.0 core:backspace (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
     -1:20.9.0 core:move-right (atom-text-editor.editor.vim-mode.is-focused.insert-mode.autocomplete-active)
     -1:20.5.0 core:move-left (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
  7x -1:03.8.0 core:backspace (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
     -0:55.1.0 core:save (atom-text-editor.editor.vim-mode.is-focused.insert-mode.autocomplete-active)
     -0:52.3.0 core:copy (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
  2x -0:50.5.0 blur (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
```

### Config

```json
{
  "core": {},
  "editor": {
    "invisibles": {},
    "showIndentGuide": true,
    "fontSize": 15
  }
}
```

### Installed Packages

```coffee
# User
autoclose-html, v0.19.0
jshint, v1.8.3
react, v0.12.10
ruby-block, v0.3.5
vim-mode, v0.64.0

# Dev
No dev packages
```

**Comment by efatsi** [comment]
Found a way to reproduce this locally and narrowed it down as best I could.

1. Create a .rb file.
2. Add exactly 41 blank lines
3. Add this to the 40th line: `apple banana apple banana apple banana apple banana apple banana apple banana apple banana`
4. Navigate cursor anywhere after the first word but before the last.
5. Hit enter.

**Comment by 50Wliu** [comment]
@efatsi Does it still reproduce in safe mode (`atom --safe`)?

**Comment by efatsi** [comment]
@50Wliu Looks like it doesn't happen in safe mode. Can check a beta release in a couple days.

**Comment by efatsi** [comment]
Fixed in 1.5.2 for me.
